ts/parts/Chart.ts

```diff
@@ -1051,119 +1059,121 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
         subtitleOptions?: Highcharts.SubtitleOptions,
         redraw?: boolean
     ): void {
-        var chart = this,
-            options = chart.options,
-            styledMode = chart.styledMode,
-            chartTitleOptions: Highcharts.TitleOptions,
-            chartSubtitleOptions: Highcharts.SubtitleOptions;
 
-        chartTitleOptions = options.title = merge(
-            // Default styles
-            (!styledMode && {
-                style: {
-                    color: '${palette.neutralColor80}',
-                    fontSize: options.isStock ? '16px' : '18px' // #2944
-                }
-            }) as Highcharts.TitleOptions,
-            options.title as Highcharts.TitleOptions,
-            titleOptions as Highcharts.TitleOptions
-        );
-        chartSubtitleOptions = options.subtitle = merge(
+        this.applyDescription('title', titleOptions);
+        this.applyDescription('subtitle', subtitleOptions);
+
+        // The initial call also adds the caption. On update, chart.update will
+        // relay to Chart.setCaption.
+        this.applyDescription('caption', undefined);
+
+        this.layOutTitles(redraw);
+    },
+
+    /**
+     * Apply a title, subtitle or caption for the chart
+     *
+     * @private
+     * @function Highcharts.Chart#applyDescription
+     *
+     * @param name {string}
+     *        Either title, subtitle or caption
+     * @param {Highcharts.TitleOptions|Highcharts.SubtitleOptions|Highcharts.CaptionOptions|undefined} explicitOptions
+     *        The options to set, will be merged with default options.
+     *
+     * @return {void}
+     */
+    applyDescription: function (
+        this: Highcharts.Chart,
+        name: ('title'|'subtitle'|'caption'),
+        explicitOptions: (Highcharts.DescriptionOptionsType|undefined)
```

Comment:
This should be written as `explicitOptions?: Highcharts.DescriptionOptionsType`.